src/node_file.cc

```diff
@@ -479,9 +479,9 @@ void FillStatsArray(double* fields, const uv_stat_t* s) {
   fields[9] = -1;
 #endif
   // Dates.
-#define X(idx, name)                          \
-  fields[idx] = (s->st_##name.tv_sec * 1e3) + \
-                (s->st_##name.tv_nsec / 1e6); \
+#define X(idx, name)                                                 \
+  fields[idx] = (static_cast<uint32_t>(s->st_##name.tv_sec) * 1e3) + \
+                (static_cast<uint32_t>(s->st_##name.tv_nsec) / 1e6); \
```

Comment:
> since casting to double is a calculation is comes out a negative double.

Well, yeah, if the `long` is negative to begin with. There’s no calculation taking place, though; if you want to convince the compiler to interpret it as a positive number, `static_cast<double>(static_cast<unsigned long>())` seems right.